keras/objectives.py

```diff
@@ -12,6 +12,9 @@ def mean_squared_error(y_true, y_pred):
 def mean_absolute_error(y_true, y_pred):
     return T.abs_(y_pred - y_true).mean(axis=-1)
 
+def mean_absolute_percentage_error(y_true, y_pred):
+    return T.abs_((y_true - y_pred) / y_true).mean() * 100
```

Comment:
Since you are taking a global mean instead of the mean on the last axis, this will fail for loss weighting.